indolaw-nextjs/components/StructureWithHeading.tsx

```diff
@@ -22,19 +29,29 @@ export default function StructureWithHeading(props: {
         }
       `}</style>
       <div id={structure.id}>
-        {structure.children
-          .slice(0, numOfHeadingLines)
-          .map((child, idx) => (
-            <PrimitiveStructure
-              key={idx}
-              structure={child as Primitive}
-              customStyle={headingStyle}
-            />
-          ))}
+        {structure.children.slice(0, numOfHeadingLines).map((child, idx) => (
+          <PrimitiveStructure
+            key={idx}
+            structure={child as Primitive}
+            customStyle={headingStyle}
+          />
+        ))}
       </div>
-      {structure.children
-        .slice(numOfHeadingLines)
-        .map((child, idx) => renderStructure(child, idx))}
+      {structure.children.slice(numOfHeadingLines).map((child, idx) => {
+        let result = [renderStructure(child, idx, penjelasanUmum)];
+
+        // If the current node is a PASAL and penjelasan umum still exists:
+        // 1. Remove and return the first node from penjelasan umum
```

Comment:
Can you elaborate in the comment why the first node needs to be removed (so we remember next time we read the code 😛)